### PR TITLE
Update FilmicSharpen, qUINT, and MXAO

### PR DESCRIPTION
**Pull request by ghost**
Ported changes from original authors:
- **FilmicSharpen:** Some GUI enhancements regarding REC 601 & 709, comments added for easier updates
- **MXAO:** Tweaks from mainstream Reshade repo, formatting fixes from your port
- Watchdogs.fx format fix
- Lightroom updates from Marty

**Note:** The authors of qUINT and FilmicSharpen have not updated their shaders for months. Therefore, these changes are here to stay for some time.

### CODE REVIEW

File: Shaders/MXAO 4.0.2 EX.fx [hunk 1]
```diff
@@ -24,75 +24,76 @@
 #endif
 
 #ifndef MXAO_SMOOTHNORMALS
- #define MXAO_SMOOTHNORMALS             1       //[0 or 1]      This feature makes low poly surfaces smoother, especially useful on older games.
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This is intentionally set to 1 in GShade, as the primary concern of the build is Final Fantasy XIV support.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 2]
```diff
@@ -24,75 +24,76 @@
...
+ #define MXAO_SMOOTHNORMALS             0       //[0 or 1]      This feature makes low poly surfaces smoother, especially useful on older games.
 #endif
 
 #ifndef MXAO_TWO_LAYER
- #define MXAO_TWO_LAYER                 1       //[0 or 1]      Splits MXAO into two separate layers that allow for both large and fine AO.
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This is intentionally set to 1 in GShade, as the primary concern of the build is Final Fantasy XIV support.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 3]
```diff
@@ -24,75 +24,76 @@
...
 
 //++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 // UI variables
 //++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 
 uniform int MXAO_GLOBAL_SAMPLE_QUALITY_PRESET <
-	ui_type = "combo";
+        ui_type = "combo";
         ui_label = "Sample Quality";
-	ui_items = "Very Low (4)\0Low (8)\0Medium (16)\0High (24)\0Very High (32)\0Ultra (64)\0Maximum (255)\0";
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx:41]: Values were added here for the sake of end-user clarity.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 4]
```diff
@@ -24,75 +24,76 @@
...
-	ui_tooltip = "Global quality control, main performance knob. Higher radii might require higher quality.";
+        ui_items = "Very Low\0Low\0Medium\0High\0Very High\0Ultra\0Maximum\0";
+        ui_tooltip = "Global quality control, main performance knob. Higher radii might require higher quality.";
 > = 2;
 
 uniform float MXAO_SAMPLE_RADIUS <
-	ui_type = "slider";
-	ui_min = 0.5; ui_max = 20.0;
+        ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 5]
```diff
@@ -24,75 +24,76 @@
...
+        ui_min = 0.5; ui_max = 20.0;
         ui_label = "Sample Radius";
-	ui_tooltip = "Sample radius of MXAO, higher means more large-scale occlusion with less fine-scale details.";
+        ui_tooltip = "Sample radius of MXAO, higher means more large-scale occlusion with less fine-scale details.";
 > = 2.5;
 
 uniform float MXAO_SAMPLE_NORMAL_BIAS <
-	ui_type = "slider";
-	ui_min = 0.0; ui_max = 2.0;
+        ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 6]
```diff
@@ -24,75 +24,76 @@
...
+        ui_min = 0.0; ui_max = 0.8;
         ui_label = "Normal Bias";
-	ui_tooltip = "Occlusion Cone bias to reduce self-occlusion of surfaces that have a low angle to each other.";
+        ui_tooltip = "Occlusion Cone bias to reduce self-occlusion of surfaces that have a low angle to each other.";
 > = 0.2;
 
 uniform float MXAO_GLOBAL_RENDER_SCALE <
-	ui_type = "slider";
+        ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 7]
```diff
@@ -24,75 +24,76 @@
...
         ui_label = "Render Size Scale";
-	ui_min = 0.50; ui_max = 1.00;
+        ui_min = 0.50; ui_max = 1.00;
         ui_tooltip = "Factor of MXAO resolution, lower values greatly reduce performance overhead but decrease quality.\n1.0 = MXAO is computed in original resolution\n0.5 = MXAO is computed in 1/2 width 1/2 height of original resolution\n...";
 > = 1.0;
 
 uniform float MXAO_SSAO_AMOUNT <
-	ui_type = "slider";
-	ui_min = 0.00; ui_max = 6.00;
+        ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 8]
```diff
@@ -24,75 +24,76 @@
...
         ui_label = "Ambient Occlusion Amount";
-	ui_tooltip = "Intensity of AO effect. Can cause pitch black clipping if set too high.";
+        ui_tooltip = "Intensity of AO effect. Can cause pitch black clipping if set too high.";
 > = 1.00;
 
 #if(MXAO_ENABLE_IL != 0)
         uniform float MXAO_SSIL_AMOUNT <
-        	ui_type = "slider";
-        	ui_min = 0.00; ui_max = 12.00;
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 9]
```diff
@@ -24,75 +24,76 @@
...
+                ui_min = 0.00; ui_max = 12.00;
                 ui_label = "Indirect Lighting Amount";
-        	ui_tooltip = "Intensity of IL effect. Can cause overexposured white spots if set too high.";
+                ui_tooltip = "Intensity of IL effect. Can cause overexposured white spots if set too high.";
         > = 4.00;
 
         uniform float MXAO_SSIL_SATURATION <
-        	ui_type = "slider";
-        	ui_min = 0.00; ui_max = 3.00;
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 10]
```diff
@@ -24,75 +24,76 @@
...
+                ui_min = 0.00; ui_max = 3.00;
                 ui_label = "Indirect Lighting Saturation";
-        	ui_tooltip = "Controls color saturation of IL effect.";
-
+                ui_tooltip = "Controls color saturation of IL effect.";
+        > = 1.00;
+        
         uniform float MXAO_SSIL_SATURATION_FILTER <
-				ui_type = "slider";
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 11]
```diff
@@ -24,75 +24,76 @@
...
                 ui_min = 0.00; ui_max = 1.00;
                 ui_label = "Indirect Lighting Saturation Filter";
                 ui_tooltip = "Controls how much unsaturated colors should be excluded from IL. Or in other words how much saturation should control the amount of light bounced. Physically inaccurate but helps reducing ugly bright corners while keeping nicer color bleeding intact.";
         > = 0.00;
         
         uniform float MXAO_SSIL_GAMMA <
-				ui_type = "slider";
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 12]
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
 
 #if (MXAO_TWO_LAYER != 0)
         uniform float MXAO_SAMPLE_RADIUS_SECONDARY <
-           	ui_type = "slider";
-           	ui_min = 0.1; ui_max = 2.00;
-                   ui_label = "Fine AO Scale";
-           	ui_tooltip = "Multiplier of Sample Radius for fine geometry. A setting of 0.5 scans the geometry at half the radius of the main AO.";
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 13]
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
...
+                ui_label = "Fine AO Scale";
+                ui_tooltip = "Multiplier of Sample Radius for fine geometry. A setting of 0.5 scans the geometry at half the radius of the main AO.";
         > = 0.2;
 
         uniform float MXAO_AMOUNT_FINE <
-             ui_type = "slider";
-             ui_min = 0.00; ui_max = 2.00;
-             ui_label = "Fine AO intensity multiplier";
-             ui_tooltip = "Intensity of small scale AO / IL.";
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 14]
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
...
+                ui_label = "Fine AO intensity multiplier";
+                ui_tooltip = "Intensity of small scale AO / IL.";
         > = 1.0;
 
         uniform float MXAO_AMOUNT_COARSE <
-             ui_type = "slider";
-             ui_min = 0.00; ui_max = 2.00;
-             ui_label = "Coarse AO intensity multiplier";
-             ui_tooltip = "Intensity of large scale AO / IL.";
+                ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 15]
```diff
@@ -101,60 +102,60 @@ uniform float MXAO_SSAO_AMOUNT <
...
+                ui_min = 0.00; ui_max = 1.00;
+                ui_label = "Coarse AO intensity multiplier";
+                ui_tooltip = "Intensity of large scale AO / IL.";
         > = 1.0;
 #endif
 
 uniform float MXAO_GAMMA <
-		ui_type = "slider";
+        ui-ui_type = "slider"
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This isn't terminated and I'm not familiar with a ui-ui-type.

File: Shaders/MXAO 4.0.2 EX.fx [hunk 16]
```diff
@@ -506,7 +507,7 @@ void PS_AmbientObscurance(in MXAO_VSOUT MXAO, out float4 color : SV_Target0)
         	#if(MXAO_ENABLE_IL != 0)
                         if(fAO > 0.1)
                         {
-        			float3 fIL = tex2Dlod(sMXAO_ColorTex, float4(sampleUV,0,sampleMIP + MXAO_MIPLEVEL_IL)).xyz;
+        			const float3 fIL = tex2Dlod(sMXAO_ColorTex, float4(sampleUV,0,sampleMIP + MXAO_MIPLEVEL_IL)).xyz;
```
Comment (Mortalitas) [inline Shaders/MXAO 4.0.2 EX.fx]: This variable cannot be const.

File: Shaders/qUINT_bloom.fx [hunk 17]
```diff
@@ -21,7 +21,7 @@
 =============================================================================*/
 
 #ifndef SAMPLE_HQ
- #define SAMPLE_HQ 1
```
Comment (Mortalitas) [inline Shaders/qUINT_bloom.fx]: This is intentionally set to 1 in GShade, as the primary concern of the build is Final Fantasy XIV support.

File: Shaders/qUINT_dof.fx [hunk 18]
```diff
@@ -122,7 +122,7 @@ uniform float fADOF_ShapeRadius <
 uniform float fADOF_SmootheningAmount <
     ui_type = "slider";
     ui_min = 0.0;
-    ui_max = 200.0;
```
Comment (Mortalitas) [inline Shaders/qUINT_dof.fx]: This was intentionally set as a cap as a user suggested it.

**Review by Mortalitas — CHANGES_REQUESTED** [review]
Some preprocessor definitions need reverting as the changes were intentional, and I think there was a find and replace issue?

**Review by Mortalitas — APPROVED** [review]

**Comment by ghost** [comment]
Ah yes that was a find and replace issue. Thanks for that

**Pull request merged**